### PR TITLE
feat: add support for experimental tooling

### LINKED ISSUE #2136: Put in place tooling around `experimental_*` features
If any used `experimental_` feature, provides a message at startup to explain and link dicussions to the corresponding feature

### PR DESCRIPTION
**Pull request by bnjjj**
close #2136

### CODE REVIEW

File: apollo-router/src/configuration/experimental.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            .into_iter()
+            .filter_map(|used_exp_conf| {
+                available_discussions
+                    .get(&used_exp_conf)
+                    .map(|discussion_link| format!("\t- {used_exp_conf}: {discussion_link}"))
+            })
+            .collect();
+        if !needed_discussions.is_empty() {
+            tracing::info!(
+                r#"You're using some "experimental" features (configuration prefixed by "experimental_"), it means we can break the configuration shape/behavior in a future release.
```
Comment (BrynCooke) [inline apollo-router/src/configuration/experimental.rs:21]:
```suggestion
                r#"You're using some "experimental" features (configuration prefixed by "experimental_"), we may make breaking changes in future releases.
```

**Review by BrynCooke — APPROVED** [review]
Nice!

File: apollo-router/src/configuration/experimental.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    format!("\t- {used_exp_conf}: {discussion_link}")
+                })
+                .collect();
+            println!(
+                "List of all experimental configurations with related GitHub discussions:\n\n{}",
+                available_exp_confs_str.join("\n")
+            );
+        }
+        Err(err) => {
+            eprintln!(
```
Comment (BrynCooke) [inline apollo-router/src/configuration/experimental.rs:23]: I think this should panic rather than print a warning. We have a malformed json file.

**Review by garypen — COMMENTED** [review]
I don't object to this, but I'd prefer something simpler. I'm not sure what the advantages are of having the experimental_features.json file are. Is it for some notional tooling thing?

File: apollo-router/src/configuration/experimental.rs (hunk 2 above)
Comment (bnjjj) [reply apollo-router/src/configuration/experimental.rs:23]: I just prefer put this kind of data in a specific file to not put some weird json directly in our source code. I also prefer write JSON in a proper .json file with syntax highlighting from my code editor, I feel better :p 
There is no other reason TBH

**Review by o0Ignition0o — APPROVED** [review]

**Review by abernix — COMMENTED** [review]
This is so awesome.  Thank you for doing this!

**Pull request merged**